### PR TITLE
First person parkour movement

### PR DESCRIPTION
**Pull request by FrankvanKan**
Added first person movement with walking, running, jumping, crouching and sliding.

Items for review are:

- `Code/Scripts/Player/*`
- `Level/Prefabs/Player`

Testable in the Parkour scene under `Level/Scenes/Parkour`

### CODE REVIEW

**Comment by SeanSnel** [comment]
[First person parkour movement](https://trello.com/c/wLbuRSc2/5-first-person-parkour-movement)

File: FP-Parkour/Assets/Code/Scripts/Player/PlayerInput.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    {
+        get
+        {
+            float sidewaysMovement = Input.GetAxis("Horizontal");
+            float forwardMovement = Input.GetAxis("Vertical");
+            Vector2 movement = new Vector2(sidewaysMovement, forwardMovement);
+            return Vector2.ClampMagnitude(movement, 1);
+        }
+    }
+
```
Comment (marthijn13) [inline FP-Parkour/Assets/Code/Scripts/Player/PlayerInput.cs]: No need for both GetAxis and GetAxisRaw

File: FP-Parkour/Assets/Code/Scripts/Player/PlayerMovement.cs [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+    private float walkSpeed = 5f;
+    [SerializeField]
+    [Tooltip("Maximum speed while sprinting")]
+    private float sprintSpeed = 7f;
+    [Header("Crouching and Sliding")]
+    [SerializeField]
+    [Tooltip("Maximum speed while crouching")]
+    private float crouchSpeed = 2f;
+    [SerializeField]
+    [Tooltip("Iinitial speed when sliding")]
```
Comment (marthijn13) [inline FP-Parkour/Assets/Code/Scripts/Player/PlayerMovement.cs]: Spelling error iinitial

File: FP-Parkour/Assets/Code/Scripts/Player/PlayerMovement.cs [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+                case PlayerStatus.SLIDING:
+                    return initialSlideSpeed;
+                default:
+                    return sprintSpeed;
+            }
+        }
+    }
+
+    [Header("Vertical Speed")]
+    [SerializeField] [Tooltip("The height that the player should jump in Unity units (usually 1unit = 1meter)")]
```
Comment (BBrendeke) [inline FP-Parkour/Assets/Code/Scripts/Player/PlayerMovement.cs]: Wees consistent met entertjes tussen prefixes, doe maar alles onder elkaar

File: FP-Parkour/Assets/Code/Scripts/Player/PlayerMovement.cs [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+    {
+        CheckCrouch();
+        MoveDefault();
+        Jump();
+        ApplyDesiredVelocity();
+    }
+
+
+    /***************************************************/
+    /******************** CROUCHING ********************/
```
Comment (BBrendeke) [inline FP-Parkour/Assets/Code/Scripts/Player/PlayerMovement.cs:122]: Voegen deze comments iets toe? Op mijn betreft niet

File: FP-Parkour/Assets/Code/Scripts/Player/PlayerMovement.cs [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+            Slide();
+        }
+        else if (status == PlayerStatus.SLIDING && grounded)
+        {
+            Slide();
+        }
+        else
+        {
+            Ray crouchRay = new Ray(transform.position, transform.up);
+            bool cantStandUp = (controller.height < 2f && Physics.Raycast(crouchRay, 1.25f));
```
Comment (BBrendeke) [inline FP-Parkour/Assets/Code/Scripts/Player/PlayerMovement.cs]: Bool kan in if

File: FP-Parkour/Assets/Code/Scripts/Player/PlayerMovement.cs [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+        status = PlayerStatus.WALKING;
+    }
+
+    /*************************************************/
+    /******************** JUMPING ********************/
+    private void Jump()
+    {
+        Vector3 groundPosition = transform.position - new Vector3(0, controller.height / 2, 0);
+        grounded = Physics.CheckBox(groundPosition, groundCheckSize, Quaternion.identity, groundMask);
+        // grounded = Physics.CheckSphere(groundPosition, groundDistance, groundMask);
```
Comment (BBrendeke) [inline FP-Parkour/Assets/Code/Scripts/Player/PlayerMovement.cs]: Als je checkbox blijft gebruiken over checksphere kan de comment weg

**Review by BBrendeke — CHANGES_REQUESTED** [review]

File: FP-Parkour/Assets/Code/Scripts/Player/PlayerMovement.cs (hunk 6 above)
Comment (marthijn13) [inline FP-Parkour/Assets/Code/Scripts/Player/PlayerMovement.cs]: code comment

**Review by marthijn13 — CHANGES_REQUESTED** [review]

**Review by marthijn13 — APPROVED** [review]

**Review by BBrendeke — APPROVED** [review]

**Pull request merged**